nucleus/payara-modules/healthcheck-core/src/main/java/fish/payara/nucleus/healthcheck/HealthCheckService.java

```diff
@@ -208,13 +209,16 @@ public void bootstrapHealthCheck() {
     /**
      * Starts all notifiers that have been enable with the healthcheck service.
      */
-    public void bootstrapNotifierList() {
+    public synchronized void bootstrapNotifierList() {
         notifierExecutionOptionsList = new ArrayList<>();
         if (configuration.getNotifierList() != null) {
             for (Notifier notifier : configuration.getNotifierList()) {
                 ConfigView view = ConfigSupport.getImpl(notifier);
                 NotifierConfigurationType annotation = view.getProxyType().getAnnotation(NotifierConfigurationType.class);
-                notifierExecutionOptionsList.add(executionOptionsFactoryStore.get(annotation.type()).build(notifier));
+                NotifierExecutionOptionsFactory<Notifier> factory = executionOptionsFactoryStore.get(annotation.type());
+                if (factory != null) {
```

Comment:
The factory can be null depending on prior changes to the notifier configurations and reboots occurring.

I'm not 100% sure this is the right fix. Maybe the factory should never be null but it solves the immediate problem.